numpy/core/src/npysort/heapsort.c.src

```diff
@@ -182,7 +183,11 @@ heapsort_@suff@(void *start, npy_intp n, void *varr)
 {
     PyArrayObject *arr = varr;
     size_t len = PyArray_ITEMSIZE(arr)/sizeof(@type@);
+#ifdef __MVS__
+    @type@ *tmp = malloc(MAX(1, PyArray_ITEMSIZE(arr)));
+#else
     @type@ *tmp = malloc(PyArray_ITEMSIZE(arr));
+#endif
```

Comment:
Looks like `heapsort` is missing these lines:

https://github.com/numpy/numpy/blob/f49ed541248bbccd0b5389fd24d13bdbf045b589/numpy/core/src/npysort/quicksort.c.src#L453-L456